web/src/toga_web/app.py

```diff
@@ -23,7 +23,7 @@ def create(self):
         self.interface.commands.add(
             # ---- Help menu ----------------------------------
             toga.Command(
-                lambda _: self.interface.about(),
```

Comment:
I don't follow this change. Previously it was taking one argument (the Command), and ignoring it. The other backends do the same thing. Now it's taking two arguments (event, widget)? How does that fit the protocol?

Reply:
Huh... I made the change to get it to work, but didn't mentally catch that the resulting method was no longer following the same protocol. I'll dig deeper into what is going on.

Reply:
🤦 

The `on_click` handler needs to be a DOM handler; it was being assigned the raw Toga handler. It worked previously because the first argument to the toga handler was eaten as part of the binding process, so the action would be invoked with the DOM event as the "widget". With the removal of the redundant argument, it was no longer being consumed. 

Fixed by making the on_click handler an actual DOM event handler, which proxies to activate the action.

The same pattern is used by Winforms and GTK, so I've modified the code to use a consistent implementation approach. Android and Cocoa use a different approach - there's a central "menu item selected" event which triggers a lookup and activation of the appropriate action.